json-rpc/src/methods.rs

```diff
@@ -120,12 +121,60 @@ impl JsonRpcRequest {
     fn version(&self) -> u64 {
         self.ledger_info.ledger_info().version()
     }
+
+    fn parse_submit_signed_transaction(
+        &self,
+        index: usize,
+        name: &str,
+    ) -> Result<SignedTransaction> {
+        let payload: String = self.parse_param(index, name)?;
+        Ok(SignedTransaction::try_from(payload).map_err(|_| {
+            JsonRpcError::invalid_param(
+                index,
+                name,
+                "hex-encoded string of LCS serialized Libra SignedTransaction type",
+            )
+        })?)
+    }
+
+    fn parse_account_address(&self, index: usize) -> Result<AccountAddress> {
+        let name = "account address";
+        let address: String = self.parse_param(index, name)?;
+        Ok(AccountAddress::from_str(&address)
+            .map_err(|_| JsonRpcError::invalid_param(index, name, "hex-encoded string"))?)
+    }
+
+    fn parse_event_key(&self, index: usize, name: &str) -> Result<EventKey> {
+        let raw_event_key: String = self.parse_param(index, name)?;
+        Ok(EventKey::try_from(raw_event_key)
+            .map_err(|_| JsonRpcError::invalid_param(index, name, "hex-encoded string"))?)
+    }
+
+    fn parse_param<T>(&self, index: usize, name: &str) -> Result<T>
+    where
+        T: DeserializeOwned,
+    {
+        Ok(serde_json::from_value(self.get_param(index)).map_err(|_| {
+            let type_info = match name {
+                "start" => "unsigned int64",
+                "start_version" => "unsigned int64",
+                "limit" => "unsigned int64",
+                "account sequence number" => "unsigned int64",
+                "include_events" => "boolean",
+                "account address" => "hex-encoded string",
+                "event key" => "hex-encoded string",
+                "known version" => "unsigned int64",
+                _ => "unknown",
+            };
+            JsonRpcError::invalid_param(index, name, type_info)
+        })?)
+    }
 }
 
 /// Submits transaction to full node
 async fn submit(mut service: JsonRpcService, request: JsonRpcRequest) -> Result<()> {
-    let txn_payload: String = serde_json::from_value(request.get_param(0))?;
-    let transaction: SignedTransaction = lcs::from_bytes(&hex::decode(txn_payload)?)?;
+    let transaction: SignedTransaction = request.parse_submit_signed_transaction(0, "data")?;
```

Comment:
no need for `:SignedTransaction` here

Reply:
updated added try_parse_from generic type instead, so keep this